src/components/fullcontact/fullcontact.js

```diff
@@ -1,21 +1,53 @@
+import { mapActions } from 'vuex'
+
 export default {
   name: 'fullcontact',
   components: {},
-  props: [
-    'contact'
-  ],
+  props: {
+    editable: {
+      type: Boolean,
+      default: false
+    },
+    contact: {
+      type: Object,
+      default: {}
+    }
+  },
   data () {
     return {
-
+      edit: this.editable,
+      currentContact: {},
+      birthdateDialog: false,
+      nameRules: [
+        v => !!v || 'Name is required',
+        v => (v && v.length <= 100) || 'Name must be less than 100 characters'
+      ],
+      emailRules: [
+        v => !!v || 'E-mail is required',
+        v => /.+@.+\..+/.test(v) || 'E-mail must be valid'
```

Comment:
Isn't there some predefined function for this?

Reply:
no